ocaml/typing/typeopt.mli

```diff
@@ -30,17 +30,30 @@ val bigarray_type_kind_and_layout :
       Env.t -> Types.type_expr -> Lambda.bigarray_kind * Lambda.bigarray_layout
 
 (* CR layouts: `layout` should have a `sort` argument.  *)
-(* CR layouts v2: [layout], [function_return_layout] and
-   [function2_return_layout] have had location arguments added just to support
-   the void check error message.  These arguments can be removed when we're
-   happy to take that check out. *)
-val layout : Env.t -> Location.t -> Types.type_expr -> Lambda.layout
+(* CR layouts v2: [layout], [function_return_layout], [function2_return_layout],
+   and [layout_of_sort] have had location arguments added just to support the
+   void check error message.  These arguments can be removed when we're happy to
+   take that check out. *)
+val layout :
+  Env.t -> Location.t -> Layouts.sort -> Types.type_expr -> Lambda.layout
+
+(* This translates a type system sort to a lambda layout.  Use the function
+   [layout] instead when the type is known: that function gives a more precise
+   result.  This should only be used when the precise layout isn't needed for
+   optimization. *)
+val layout_of_sort : Location.t -> Layouts.sort -> Lambda.layout
 
 val function_return_layout :
-  Env.t -> Location.t -> Types.type_expr -> Lambda.layout
+  Env.t -> Location.t -> Layouts.sort -> Types.type_expr -> Lambda.layout
```

Comment:
Please add comments to these functions describing that the sort is _not_ the sort of the type passed. This is potentially confusing!